platform/linuxbsd/display_server_x11.cpp

```diff
@@ -3603,8 +3603,11 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
 		}
 	}
 #endif
-
-	WindowID main_window = _create_window(p_mode, p_flags, Rect2i(Point2(), p_resolution));
+	WindowData &main_window = windows[MAIN_WINDOW_ID];
+	Point2i window_position(
+			(main_window.size.width - p_resolution.x) / 2,
+			(main_window.size.height - p_resolution.y) / 2);
```

Comment:
Same here main_window is create later, should be the same code as on macOS.

And probably the same on Windows too, just in case we'll need to create main window on non-primary monitor or add some DPI scaling to the `screen_get_` functions.